File: AddOns/Share/Categories/TripRequest+Shareable.m
//
//  TripRequest+Shareable.m
//  Pods
//
//  Created by Adrian Schoenig on 24/06/2016.
//
//

#import "TripRequest+Shareable.h"

#import "TKShareHelper.h"

@implementation TripRequest (Shareable)

- (NSURL *)shareURL
{
  return [TKShareHelper queryURLForStart:[self.fromLocation coordinate]
                                     end:[self.toLocation coordinate]
                                timeType:self.type
                                    time:self.time];
}


@end

File: AddOns/Share/Categories/TripRequest+Shareable.swift
//
//  TripRequest+Shareable.swift
//  TripKit
//
//  Created by Adrian Schoenig on 30/6/17.
//  Copyright © 2017 SkedGo. All rights reserved.
//

import Foundation

extension TripRequest: TKURLShareable {
  
  public var shareURL: URL? {
    return TKShareHelper.createQueryURL(start: fromLocation.coordinate, end: toLocation.coordinate, timeType: type, time: time)
  }
  
}
